### PR TITLE
Backporting changes from Private Datasets feature branch

### PR DESCRIPTION
**Pull request by s373r**
## Description

- Some changes are piling up in the Private Datasets branch that are better put in before the feature is finalized.
- Also, it will help for readers of the Private Datasets branch to not be distracted by changes that are not directly related to the epic

### Added
- kamu-adapter-graphql: added macros (`from_catalog_n!()` & `unsafe_from_catalog_n!()`) 
   that simplify the extraction of components from the DI catalog
- database-common: the logic for pagination of data processing is generalized in `EntityPageStreamer`
### Changed
- Speed up project build time by removing unused dependencies which were not detected by automated tools

## Checklist before requesting a review

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [ ] Observability:
    
  - [ ] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ✅
    
  - [ ] Alerts: ✅
- [ ] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [ ] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅
    
  - [ ] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ✅
    
  - [ ] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1)

### CODE REVIEW

File: resources/openapi-mt.json [hunk 1]
```diff
@@ -1398,7 +1398,7 @@
...
... \"f1620e91f4d3fa26bc4ca0c49d681c8b630550239b64d3cbcfd7c6c2d6ff45998b088\",\n        \"subQueriesHash\": \"f1620ca4510738395af1429224dd785675309c344b2b549632e20275c69b15ed1d210\"\n    },\n    \"proof\": {\n        \"type\": \"Ed25519Signature2020\",\n        \"verificationMethod\": \"did:key:z6MkkhJQPHpA41mTPLFgBeygnjeeADUSwuGDoF9pbGQsfwZp\",\n        \"proofValue\": \"uJfY3_g03WbmqlQG8TL-WUxKYU8ZoJaP14MzOzbnJedNiu7jpoKnCTNnDI3TYuaXv89vKlirlGs-5AN06mBseCg\"\n    }\n}\n```\n\nA client that gets a proof in response should\nperform [a few basic steps](https://docs.kamu.dev/node/commitments#response-validation) to validate\nthe proof integrity. For example making sure that the DID in\n`proof.verificationMethod` actually corresponds to the node you're querying\ndata from and that the signature in `proof.proofValue` is actually valid.\nOnly after this you can use this proof to hold the node accountable for the\nresult.\n\nA proof can be stored long-term and then disputed at a later point using\nyour own node or a 3rd party node you can trust via the\n[`/verify`](#tag/odf-query/POST/verify) endpoint.\n\nSee [commitments documentation](https://docs.kamu.dev/node/commitments) for details.",
```
Comment (s373r) [inline resources/openapi-mt.json:1401]: Note for readers: 
image

File: resources/openapi.json [hunk 2]
```diff
@@ -1367,7 +1367,7 @@
...
... \"f1620e91f4d3fa26bc4ca0c49d681c8b630550239b64d3cbcfd7c6c2d6ff45998b088\",\n        \"subQueriesHash\": \"f1620ca4510738395af1429224dd785675309c344b2b549632e20275c69b15ed1d210\"\n    },\n    \"proof\": {\n        \"type\": \"Ed25519Signature2020\",\n        \"verificationMethod\": \"did:key:z6MkkhJQPHpA41mTPLFgBeygnjeeADUSwuGDoF9pbGQsfwZp\",\n        \"proofValue\": \"uJfY3_g03WbmqlQG8TL-WUxKYU8ZoJaP14MzOzbnJedNiu7jpoKnCTNnDI3TYuaXv89vKlirlGs-5AN06mBseCg\"\n    }\n}\n```\n\nA client that gets a proof in response should\nperform [a few basic steps](https://docs.kamu.dev/node/commitments#response-validation) to validate\nthe proof integrity. For example making sure that the DID in\n`proof.verificationMethod` actually corresponds to the node you're querying\ndata from and that the signature in `proof.proofValue` is actually valid.\nOnly after this you can use this proof to hold the node accountable for the\nresult.\n\nA proof can be stored long-term and then disputed at a later point using\nyour own node or a 3rd party node you can trust via the\n[`/verify`](#tag/odf-query/POST/verify) endpoint.\n\nSee [commitments documentation](https://docs.kamu.dev/node/commitments) for details.",
```
Comment (s373r) [inline resources/openapi.json:1370]: ditto

File: src/adapter/graphql/src/scalars/account.rs [hunk 3]
```diff
@@ -40,7 +40,7 @@ impl From<&AccountID> for odf::AccountID {
 
 impl From<AccountID> for String {
     fn from(val: AccountID) -> Self {
-        val.0.as_did_str().to_string()
+        val.0.to_string()
```
Comment (s373r) [inline src/adapter/graphql/src/scalars/account.rs:43]: Note for readers: the same but shorter (here and below)

File: src/adapter/graphql/tests/tests/test_error_handling.rs [hunk 4]
```diff
@@ -101,27 +105,4 @@ async fn test_internal_error() {
...
-//   libunwind: stepWithCompactEncoding - invalid compact unwind encoding
-#[cfg(not(target_os = "macos"))]
-#[test_log::test(tokio::test)]
-// We use the substring part because we have a dynamic panic message part.
-#[should_panic(expected = "called `Result::unwrap()` on an `Err` value: \
-                           Unregistered(UnregisteredTypeError { type_id: TypeId { t: ")]
-async fn test_handler_panics() {
-    // Not expecting panic to be trapped - that's the job of an HTTP server
-    let schema = kamu_adapter_graphql::schema_quiet();
-    schema.execute(async_graphql::Request::new(indoc!(
```
Comment (s373r) [inline src/adapter/graphql/tests/tests/test_error_handling.rs:115]: Note for readers: now we get the error, not the panic

Comment (s373r) [reply src/adapter/graphql/tests/tests/test_error_handling.rs:115]: For this reason, I have decided to remove this test

**Review by s373r — COMMENTED** [review]
Added notes for readers

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**